govtool/frontend/src/utils/drepSearchPhraseProcessor.ts

```diff
@@ -0,0 +1,33 @@
+import { decodeCIP129Identifier } from "./cip129identifier";
+
+/**
+ * Processes the search phrase for dRep and returns the dRep ID.
+ * If the phrase starts with "drep_script" or "drep",
+ * it decodes the CIP129 identifier and extracts the transaction ID.
+ * If the transaction ID starts with "22" or "23", it returns the ID without the prefix.
+ * If any error occurs during processing, it returns the original phrase.
+ *
+ * @param phrase - The search phrase to be processed.
+ * @returns The dRep ID extracted from the search phrase or the original phrase if an error occurs.
+ */
+export const dRepSearchPhraseProcessor = async (phrase: string) => {
+  let drepIDPhrase = phrase;
+
+  try {
+    if (
+      drepIDPhrase.startsWith("drep_script") ||
+      drepIDPhrase.startsWith("drep")
+    ) {
+      const { txID } = decodeCIP129Identifier(drepIDPhrase);
+
+      drepIDPhrase = txID;
+    }
+    if (drepIDPhrase.startsWith("22") || drepIDPhrase.startsWith("23")) {
```

Comment:
Couldnt this also catch CIP-105 DRep IDs which start with `22` or `23`?

To differentiate between 105 and 129 IDs, you can just do a length check, as 129 are always longer

Reply:
It just cuts the byte from the raw DRep ID, it doesn't matter if it's CIP-105 or 129

Reply:
okay cool, thanks